src/main/java/com/avail/interpreter/levelTwo/operation/L2_MAKE_IMMUTABLE.java

```diff
@@ -142,6 +183,29 @@ public void toString (
 		builder.append(read.registerString());
 	}
 
+	/**
+	 * {@code L2_MAKE_IMMUTABLE} is idempotent, so we have a choice whether to
+	 * introduce a partial redundancy.  An easy answer is to check if the result
+	 * is ever used outside a reification {@link Zone}, and if so, avoid the
+	 * creation of the redundancy.  Otherwise all uses are in reification zones,
+	 * which are low-frequency code, so it doesn't matter much if we introduce
+	 * some redundancy along paths running through these zones.
+	 *
+	 * @param instruction
+	 *        The {@link L2Instruction} using this operation.
+	 * @return Whether to replicate this instruction into multiple successor
+	 *         blocks, even if some successors have multiple incoming edges
+	 *         (and which might not need the computation).
+	 */
+	@Override
+	public boolean shouldReplicateIdempotently (final L2Instruction instruction)
+	{
+		return true;
+		// TODO Does this need to adjust the type in the manifest of the edges
+		// it's moving across, so placeholder-replacement code won't use the
+		// old, not necessarily immutable values?
```

Comment:
What's the status of this?

Reply:
Deleted method.  This mechanism was an unsuccessful experiment.